tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.oldtest;
+
+import net.neoforged.fml.common.Mod;
+
+public class OrderedModLoadingTest {
+    private static final Object lock = new Object();
+    private static int constructedMod = 0;
+
+    private static void onConstruct(int index) {
+        synchronized (lock) {
+            if (constructedMod != index - 1) {
+                throw new IllegalStateException("Mod " + constructedMod + " was constructed, cannot construct " + index);
+            }
+        }
+
+        try {
+            Thread.sleep(10);
```

Comment:
Does the delay serve any purpose?

Reply:
Yes, it ensures that we can detect with high probability any attempt at scheduling mod N+1 before mod N is finished completing. FYI these are milliseconds not seconds.

Reply:
That feels like a pretty flaky approach to me, but fine.

Reply:
What else would you suggest? Multi-threading is inherently non-deterministic. I will note that this code managed to crash when the sorting logic was not implemented correctly in FML every single time, so at least it works. :P

Reply:
I would have made the timeout longer (maybe 1 second) to minimize flakiness from the OS being busy, but then you would have to reduce the number of mods to make the test not take too long to run.

Reply:
I feel like 5 is already not that much. 😅 Well this is probably fine as-is anyway.